### DISCUSSION TITLE
Increase supported file size for multipart/form-data

### DISCUSSION THREAD

**Discussion by I3enito** [vercel/next.js#15745] [Help] [answered]
# Feature request

## Is your feature request related to a problem? Please describe.

I use a serverless function to upload images to s3. Because it handles multipart/form-data I disabled the bodyparser in the config. Everything works fine both locally and in the vercel deployment, however if I upload an image which is bigger than 5mb I get an 413 error. This only happens on vercel, locally I don't run in this problem. 

`{"error": {"code": "413", "message": "Request Entity Too Large"}}`

The failing requests aren't logged in the vercel dashboard, so I assume they get blocked somewhere before the function even gets started.

Below the important parts of my code

```ts
import nextConnect from 'next-connect'
import multer from 'multer'
import S3 from 'aws-sdk/clients/s3'
import { NowRequest, NowResponse } from '@vercel/node'

type NowRequestWithFile = NowRequest & {
    file: Express.Multer.File
}

// initialize api-route handler
const handler = nextConnect()

// configure multer for file upload
// store uploaded image on temporary directory of serverless function
const upload = multer({ dest: '/tmp' })

const s3 = new S3({
...
})

// use multer middleware
handler.use(upload.single('file'))

handler.post(async (req: NowRequestWithFile, res: NowResponse) => {
/*... function body... */
})

export const config = {
    api: {
        bodyParser: false,
    },
}

export default handler
```

## Describe the solution you'd like

I would like to have a way to increase the accepted request size for this function.

## Describe alternatives you've considered

I could achieve the desired functionality with a dedicated node server.

**Comment by paulogdm** [comment, marked as answer]
I recommend that you stream the file directly to S3 with a "signed POST" request. You can read more about our limits [here](https://vercel.com/docs/v2/platform/limits#serverless-function-payload-size-limit). Serverless Functions are unsuited for large payloads, unfortunately.

Here is one example: https://github.com/paulogdm/s3-example/blob/master/api/helpers/s3.js

**Reply by I3enito** [reply]
Thanks for the quick response.

**Reply by aorsten** [reply]
@paulogdm is there a similar example for azure storage anywhere?

It would, by the way, be great to get a warning about this when pushing an endpoint with `sizeLimit` higher than 5mb. I've pushed 15mb as limit to some endpoints in my project, which is in prod, but luckily didn't come across this bug until I read it here.

**Reply by paulogdm** [reply]
@aorsten Not that I'm aware of. It can be a good idea to simply host your backend elsewhere. Vercel is a good platform for frontends and Serverless Functions as helpers.

You do need to take a look at our documentation: https://vercel.com/docs/v2/platform/limits?query=limits#serverless-function-payload-size-limit

**Comment by julijatrescenko** [comment]
I came across this issue when I wanted to submit an attachment to the Sengrid and send an email. Altho the Sengrid allows up to 30Mb message size. So this would be still helpful to control the file size.
